File: Patio-El-Olvidado/src/types/pedido.ts
export type PedidoTipo = 'Local' | 'ParaLlevar'
export type PedidoEstado = 'EnPreparacion' | 'Listo' | 'Entregado' | 'Cancelado'

export interface DetallePedido {
  id: number
  productoId: number
  productoNombre: string | null
  cantidad: number
  precioUnitario: number
  subtotal: number
}

export interface Pedido {
  id: number
  tipo: PedidoTipo | string
  estado: PedidoEstado | string
  subtotal: number
  total: number
  descuentoMonto?: number
  descuentoAplicado?: boolean
  fechaCreacion: string
  clienteId: number | null
  creadoPorUsuarioId: number
  detalles: DetallePedido[]
}

export interface PedidoFilter {
  estado?: string
  tipo?: string
  desde?: string
  hasta?: string
}

export interface DetallePedidoLinePayload {
  productoId: number
  cantidad: number
}

export interface PedidoPayload {
  tipo: PedidoTipo
  clienteId?: number | null
  detalles: DetallePedidoLinePayload[]
}

export interface CambiarEstadoPayload {
  estado: Exclude<PedidoEstado, 'EnPreparacion'>
}

File: API-Patio-El-Olvidado/src/PatioElOlvidado.Domain/Entities/Pedido.cs
namespace PatioElOlvidado.Domain.Entities;

public class Pedido
{
    public int Id { get; set; }
    /// <summary>Local | ParaLlevar</summary>
    public string Tipo { get; set; } = string.Empty;
    /// <summary>EnPreparacion | Listo | Entregado | Cancelado</summary>
    public string Estado { get; set; } = string.Empty;
    public decimal Subtotal { get; set; }
    public decimal Total { get; set; }
    /// <summary>Opcional: pedido anónimo / sin fidelizar.</summary>
    public int? ClienteId { get; set; }
    /// <summary>True cuando el cobro completo ya incrementó Cliente.Visitas (idempotente).</summary>
    public bool VisitaContabilizada { get; set; }
    public int CreadoPorUsuarioId { get; set; }
    public DateTime FechaCreacion { get; set; }

    public Cliente? Cliente { get; set; }
    public Usuario CreadoPorUsuario { get; set; } = null!;
    public ICollection<DetallePedido> Detalles { get; set; } = new List<DetallePedido>();
    public ICollection<Pago> Pagos { get; set; } = new List<Pago>();
}
